src/Orc.NuGetExplorer/Cache/IconCache.cs

```diff
@@ -84,19 +94,27 @@ private BitmapImage CreateImage(Stream stream)
 
         private BitmapImage CreateImage(Uri uri)
         {
+            ArgumentNullException.ThrowIfNull(uri);
+
             var image = new BitmapImage();
 
             // Find extracted resource in folder from uri
             var iconUri = uri;
 
             if (!string.IsNullOrEmpty(uri.Fragment))
             {
-                var fileName = uri.Fragment.Substring(1, uri.Fragment.Length - 1);
+                var fileName = uri.Fragment[1..];
                 var folderPath = Path.GetDirectoryName(uri.AbsolutePath);
-
-                // Decode spaces
-                folderPath = folderPath.Replace("%20", " ");
-                iconUri = new Uri(Path.Combine(folderPath, fileName));
+                if (string.IsNullOrEmpty(folderPath))
+                {
+                    Log.Debug("Cannot parse source uri for local icon");
+                }
+                else
+                {
+                    // Decode spaces
+                    folderPath = folderPath.Replace("%20", " ");
```

Comment:
I think there is a better way to escape more than just %20, can't remember the class name but there is a utility class for this.